api/utils/error_parsing.ts

```diff
@@ -0,0 +1,100 @@
+interface IErrorLocation {
+  line?: number;
+  column?: number;
+}
+
+interface IAzureEvalError {
+  originalError?: {
+    info?: {
+      message?: string;
+    };
+  };
+  errorLocation?: IErrorLocation;
+}
+
+export const AzureEvalErrorParser = (error: IAzureEvalError) => {
+  // expected error format:
+  // // Parse error at line: 14, column: 13: Incorrect syntax near 'current_date'.
+  if (error.originalError && error.originalError.info && error.originalError.info.message) {
+    const message = error.originalError.info.message;
+    // extract all characters between line: and , inclusive
+    const lineMatch = message.match(/line:(.*?),/g);
+    // if we get more than one match, bail out
+    if (lineMatch.length !== 1) {
+      return error;
+    }
+    // get rid of `line: `
+    const lineLess = lineMatch[0].replace(/[line: ]/g, "");
+    // get rid of trailing comma
+    const lineNumber = lineLess.replace(/[,]/g, "");
+
+    // extract all characters between column: and , inclusive
+    const columnMatch = message.match(/column:(.*?):/g);
+    // if we get more than one match, bail out, but keep the line number
+    if (columnMatch.length !== 1) {
+      error.errorLocation = { line: Number(lineNumber) };
+      return error;
+    }
+    // get rid of `column: `
+    const columnLess = columnMatch[0].replace(/[column: ]/g, "");
+    // get rid of trailing :
+    const columnNumber = columnLess.replace(/[:]/g, "");
+
+    error.errorLocation = { line: Number(lineNumber), column: Number(columnNumber) };
+  }
+  return error;
+};
+
+interface IRedshiftEvalError {
+  message: string;
+  position?: string;
+  errorLocation?: IErrorLocation;
+}
+
+export const RedshiftEvalErrorParser = (statement: string, error: IRedshiftEvalError) => {
+  // expected error format:
+  // // e.position = "123" - position is the number of characters into the query that the error was found at
+  // // including \n characters
+
+  if (!error.position) {
+    return error;
+  }
+  // split statement into lines
+  const statementSplitByLine = statement.split("\n");
```

Comment:
i think this logic is a little confusing, and i worry that it's calculating the wrong number (by removing these characters). how about something like:

```
const statementBeforeError = statement.substring(0, error.position);
const lineNumber = statementBeforeError.split("\n").length;
const lineIncludingError = statementBeforeError.split("\n").slice("-1")[0];
const colNumber = lineIncludingError.length;
```